python_ta/transforms/ExprWrapper.py

```diff
@@ -189,6 +192,97 @@ def parse_container_op(
         """Convert an astroid List, Set, Tuple node to a list of z3 expressions."""
         return [self.reduce(element) for element in node.elts]
 
+    def apply_in_op(
+        self, left: z3.ExprRef, right: Union[z3.ExprRef, List[z3.ExprRef], str], negate=False
+    ) -> z3.ExprRef:
+        """
+        Apply `in` or `not in` operator on a list or string and return the
+        resulting z3 expression. Raise Z3ParseException if the operands
+        do not support `in` operator
+        """
+        if isinstance(right, list):  # container type (list/set/tuple)
+            return (
+                z3.And(*[left != element for element in right])
+                if negate
+                else z3.Or(*[left == element for element in right])
+            )
+        elif isinstance(right, (str, z3.SeqRef)):  # string literal or variable
+            return z3.Not(z3.Contains(right, left)) if negate else z3.Contains(right, left)
+        else:
+            op = "not in" if negate else "in"
+            raise Z3ParseException(
+                f"Unhandled binary operation {op} with operator types {left} and {right}."
+            )
+
+    def _parse_number_literal(self, node) -> Optional:
+        """
+        If the subtree from `node` represent a number literal, return the value
+        Otherwise, return None
+        """
+        # positive number
+        if isinstance(node, nodes.Const) and isinstance(node.value, (int, float)):
+            return node.value
+        # negative number
+        elif (
+            isinstance(node, nodes.UnaryOp)
+            and node.op == "-"
+            and isinstance(node.operand, nodes.Const)
+            and isinstance(node.operand.value, (int, float))
+        ):
+            return -node.operand.value
+        else:
+            return None
+
+    def parse_subscript_op(self, node: nodes.Subscript) -> z3.ExprRef:
+        """
+        Convert an astroid Subscript node to z3 expression.
+        This method only supports string values and integer literal (both positive and negative) indexes
+        """
+        value = self.reduce(node.value)
+        if isinstance(value, z3.SeqRef):
+            slice = node.slice
+
+            # handle indexing
+            index = self._parse_number_literal(slice)
+            if isinstance(index, int):
+                return z3.SubString(value, index, index)
```

Comment:
I think you're misunderstanding the interface of [`z3.SubString`](https://z3prover.github.io/api/html/namespacez3py.html#a6d44d359a0f076c4a3e4c4af6d7f6994). Look at the name of the third parameter.